bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/FeatureInstaller.java

```diff
@@ -389,65 +392,90 @@ private synchronized void installFeatures(FeaturesService featuresService, Set<S
             featuresService.installFeatures(addons,
                     EnumSet.of(FeaturesService.Option.Upgrade, FeaturesService.Option.NoFailOnFeatureNotFound));
             logger.debug("Installed '{}'", StringUtils.join(addons, ", "));
-            for (String addon : addons) {
-                if (isInstalled(featuresService, addon)) {
-                    postInstalledEvent(addon);
+            try {
+                Feature[] features = featuresService.listInstalledFeatures();
+                for (String addon : addons) {
+                    if (isInstalled(features, addon)) {
+                        postInstalledEvent(addon);
+                    }
                 }
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception